solr/bin/solr

```diff
@@ -1973,7 +1973,7 @@ fi
 
 LOG4J_CONFIG=()
 if [ -n "$LOG4J_PROPS" ]; then
-  LOG4J_CONFIG+=("-Dlog4j.configurationFile=$LOG4J_PROPS")
+  LOG4J_CONFIG+=("-Dlog4j.configurationFile=$LOG4J_PROPS -Dlog4j2.is.webapp=false -Dlog4j2.garbagefreeThreadContextMap=true -Dlog4j2.enableDirectEncoders=true -Dlog4j2.enable.threadlocals=true")
```

Comment:
Can you add a comment with a link to some docs or anything else explaining where these properties come from and how we found them? I'm trying to search for them and coming up short. Maybe I'll dive into log4j code base next, although I'd like to avoid that if I can.

Reply:
details in jira here: https://issues.apache.org/jira/browse/SOLR-15477